nodes/src/nodes/llm_base/streaming.py

```
@@ -0,0 +1,377 @@
+# ---------------------------------------------------------------------------
+
+
+class _InstanceLike(Protocol):
+    """Minimal interface expected from the ``IInstance`` reference.
+
+    Only ``instance.sendSSE`` is required.  This keeps the module
+    importable in test environments that don't have ``engLib``.
+    """
+
+    class _Instance(Protocol):
+        def sendSSE(self, type: str, **data) -> None: ...
```

Ruff (0.15.7)

[error] Function argument `type` is shadowing a Python builtin

(A002)